lib/awful/widget/inputbox.lua

```diff
@@ -0,0 +1,530 @@
+    ret._private = {}
+    ret._private.text = args.text or ""
+    ret._private.fg = args.fg or beautiful.fg_normal
+    ret._private.border_focus_color = args.border_focus_color or beautiful.border_focus
+    ret._private.hint_text = args.inputbox_hint_text or ""
+    ret._private.hint_fg = args.hint_fg or beautiful.inputbox_hint_fg
+    ret._private.cursor_bg = args.cursor_bg or beautiful.inputbox_cursor_bg
+    ret._private.cursor_fg = args.cursor_fg or beautiful.inputbox_cursor_fg
+    ret._private.highlight_bg = args.highlight_bg or beautiful.inputbox_highlight_bg
+    ret._private.highlight_fg = args.highlight_fg or beautiful.inputbox_highlight_fg
+
+    local inputbox_widget = wibox.widget {
+        {
+            {
+                {
+                    widget = wibox.widget.textbox,
+                    text = ret._private.text,
+                    halign = args.halign or beautiful.inputbox_halign,
+                    valign = args.valign or beautiful.inputbox_valign,
+                    id = "input_text",
+                },
+                widget = wibox.container.margin,
+                margins = args.margin or beautiful.inputbox_margin,
+            },
+            widget = wibox.container.constraint,
+            strategy = "exact",
+            width = args.width or beautiful.inputbox_width,
+            height = args.height or beautiful.inputbox_height,
+        },
+        widget = wibox.container.background,
+        bg = args.bg or beautiful.bg_normal,
+        fg = args.fg or beautiful.fg_normal,
+        border_color = args.border_color or beautiful.border_color,
+        border_width = args.border_width or beautiful.border_width,
+        shape = args.shape or beautiful.shape,
+    }
+
+    ret.widget = inputbox_widget
+
+    ret.widget:buttons(
+        gtable.join {
+            awful.button({}, 1, function()
+                if not awful.keygrabber.is_running then
+                    ret:focus()
+                    ret:run(args.callbacks)
+                end
+
+                -- Stops the mousegrabber when not clicked on the widget
+                capi.mousegrabber.run(
+                    function(m)
+                        if m.buttons[1] then
+                            if capi.mouse.current_wibox ~= ret.widget then
+                                ret:emit_signal("keygrabber::stop", "")
+                                return false
+                            end
+                        end
+                        return true
+                    end,
+                    -- Remove as soon as this is optional
+                    capi.mouse.current_wibox.cursor
+                )
+            end),
+            awful.button({}, 3, function()
+                -- TODO: Figure out how to paste with highlighted support
+                -- Maybe with a signal?
+            end)
+        }
+    )
+
+    -- Change the cursor to "xterm" on hover over
+    local old_cursor, old_wibox
```

Comment:
What's `old_wibox` for?
It's set and unset in the handlers below, but never used.

Reply:
I use it to save `mouse.current_wibox` so I can set a cursor to it. I figured it would be better than to call `mouse.current_wibox` twice (e.g. `old_wibox.cursor = "xterm"`)

I could probably rename it since "old" is not too accurate in this contect, better would be "current_wibox"

Reply:
Also, `w` is the variable you save to temporarily to assign the cursor.
`old_wibox` is, as I said, never used.

Reply:
Ah, yes. Missed that line.